src/test/java/net/minecraftforge/debug/PermissionTest.java

```
@@ -0,0 +1,107 @@
+    }
+
+    @SubscribeEvent
+    public void registerCommands(RegisterCommandsEvent event)
+    {
+        event.getDispatcher().register(Commands.literal("permtest")
+            .requires(src -> canUseCommand(src, boolPerm))
+            .executes(context -> {
+                context.getSource().sendSuccess(new TextComponent("Blob"), false);
+                context.getSource().sendSuccess(new TextComponent("String:" + PermissionAPI.getPermissions((ServerPlayer) context.getSource().getEntity(), stringPerm)), false);
+                context.getSource().sendSuccess(new TextComponent("Int: " + PermissionAPI.getPermissions((ServerPlayer) context.getSource().getEntity(), intPerm)), false);
+                context.getSource().sendSuccess(PermissionAPI.getPermissions((ServerPlayer) context.getSource().getEntity(), componentPerm), false);
+
+                return 1;
+            }));
+
+        //This command should not be usable as the command isn't registered.
+        //It should also print an error
+        event.getDispatcher().register(Commands.literal("permtesterr")
+            .requires(src -> canUseCommand(src, unregisteredPerm))
+            .executes(context -> {
+                context.getSource().sendSuccess(new TextComponent("Blob"), false);
+                return 1;
+            }));
+    }
+
+    /**
+     * Required as the Default exception handling for commands fails to handle this.
+     * And produces an ambiguous error.
+     */
+    private static boolean canUseCommand(CommandSourceStack src, PermissionNode<Boolean> booleanPermission, PermissionDynamicContext<? extends StringRepresentable>... context)
+    {
+        try
+        {
+            return src.getEntity() != null && src.getEntity() instanceof ServerPlayer && PermissionAPI.getPermissions((ServerPlayer) src.getEntity(), booleanPermission, context);
+        } catch (Exception e)
+        {
+            LOGGER.error(e.getMessage());
+            return false;
+        }
+    }
```

What does this do exactly, for the record? And would it be worth exposing this as part of the public API, for modder's convenience?

---

All it does is catch the UnregisteredPermissionException, as it's otherwise swallowed by other resulting exceptions in the command tree. 

Modders shouldn't really run into this issue when using the API correctly (not forgetting to register their perms).
And they should already have some proper error handling for everything that might fail in their commands.

The method is only in the test for convenience and as a failing test case is required.

---

Sounds fine to me, though that should be explained in a comment (the javadoc).